### DISCUSSION TITLE
Question: How can I advertise a Beacon?

### DISCUSSION THREAD

**Discussion by carlosriveroib** [innoveit/react-native-ble-manager#1069] [Q&A]
Hi, I want to advertise a beacon like in this library https://github.com/jaidis/react-native-ibeacon-simulator calling a method or something like `BeaconBroadcast.startAdvertisingBeaconWithString(uuid, identifier, major, minor)`
Using that library I can do on iOS but the library is very outdated to do it with Android

Can I achieve that with this library? I really appreciate some help
Thank you very much in advance

**Comment by R3D347HR4Y** [comment]
Same question, I need this feature, please recommend another lib if this one cannot do it

**Comment by seba9999** [comment]
@carlosriveroib @R3D347HR4Y Did you suceed to do this ? With another lib ? Another way ?

**Reply by carlosriveroib** [reply]
I use a very bad workaround, but for the moment it work's for me. I use this library @criveroib/react-native-beacon-emitter which I copied from another and only works with Android. 

For iOS I use this one @criveroib/react-native-ibeacon-simulator which I forked from another and I fixed some bugs.

I would like to create a new one written in Kotlin and Swift. But I don't know them and I don't have time to do it.

**Reply by seba9999** [reply]
[@criveroib/react-native-beacon-emitter](https://www.npmjs.com/package/@criveroib/react-native-beacon-emitter) seems deprecated ... Or at least the github project was moved / deleted. [I got a 404 trying to get there](https://github.com/carlosriveroib/criveroib-react-native-beacon-emitter) !

Are you using your package : [carlosriveroib/react-native-beacon-emitter/](https://github.com/carlosriveroib/react-native-beacon-emitter/) ?

Does it work on Android  ? ( I don't care about iOS ;) )
Can I advertize a service-characteristic ? 
Am I able to write data on it then ? With this current library then ?

**Reply by carlosriveroib** [reply]
Yes, I'm using that
here's the link: https://github.com/carlosriveroib/react-native-beacon-emitter
It works on Android and with RN 0.72
I cannot answer that, I don't know about this, just I advertise a uuid that come from the back

```
const emit = (beacon: string) => { 

  if (Platform.OS === 'android') {
    await requestMultiple([
      PERMISSIONS.ANDROID.ACCESS_FINE_LOCATION,
      PERMISSIONS.ANDROID.BLUETOOTH_ADVERTISE,
      PERMISSIONS.ANDROID.BLUETOOTH_SCAN,
    ])

    stop()
    start(beacon, 65535, 0)
    setTimeout(() => {
      stop()
    }, 3000)
  }
}
```

I know is a shit but it works for my requeriments and for me it's enough

**Reply by seba9999** [reply]
Oh I see you just need some fixed ID...
In my case I would like to pass some data through differents phones ...

I think I'll surrender and go with socket.io haha ... I just wanted to bypasss some server 😢 

Thanks for your answer ! 😄

**Comment by marcosinigaglia** [comment]
You can't advertise with this library.
